pkg/compat/client.go

```diff
@@ -57,14 +59,30 @@ func NewClient(restCfg *rest.Config) (Client, error) {
 	if err != nil {
 		return nil, err
 	}
-	major, err := strconv.Atoi(version.Major)
-	if err != nil {
-		return nil, err
-	}
-	minor, err := strconv.Atoi(strings.Trim(version.Minor, "+"))
-	if err != nil {
-		return nil, err
+
+	// Attempt parsing version.Major/Minor first, fall back to parsing gitVersion
+	major, err1 := strconv.Atoi(version.Major)
+	minor, err2 := strconv.Atoi(strings.Trim(version.Minor, "+"))
+
+	if err1 != nil || err2 != nil {
+		// gitVersion format ("v1.11.0+d4cacc0")
+		r, _ := regexp.Compile(`v[0-9]+\.[0-9]+`)
```

Comment:
It's probably better to verify that there's a second "." in the version string (i.e. that there's always a third number), otherwise the "." delimited split below won't pull out just an int for the second value:
`regexp.Compile(`v[0-9]+\.[0-9]+\.`)`

Reply:
ACK, good catch! I'll change this and re-test on a few sample `gitVersion` strings before merging.